### ISSUE TITLE
package manager use case: a Zig module can express it needs to be linked against a C library artifact

### ISSUE THREAD

**Issue by slimsag** [ziglang/zig#16206]
### Zig Version

0.11.0-dev.3803+7ad104227

### Steps to Reproduce and Observed Behavior

I wanted to file an issue for a use-case that the package manager doesn't really support right now. It's what I talked to you about at SYCL @andrewrk - #14731 would solve this, I just wanted to communicate it more concisely and with a minimal-ish repro.

Here's the situation:

1. ✅ [basisu](https://github.com/hexops/basisu/blob/3a691fda2bddd8d90a22044f0dca9bb87e05197c/build.zig): a C library is packaged with the Zig build system; it exposes a static library artifact named `"basisu"`
2. ✅ [basisu-zig](https://github.com/slimsag/basisu-zig/blob/main/build.zig): a Zig wrapper for the C library above, it provides:
    * its own _static library artifact_ named `"basisu-zig"` (it has to, because `Build.Module` does not have an `addCSourceFiles` method - a related issue.)
    * a _Zig module_ named `"basisu-zig"`
        * ..which must link against the `basisu` C library artifact via the package manager:

```zig
lib.linkLibrary(b.dependency("basisu").artifact("basisu"));
```

3. ❌ [basisu-zig-example](https://github.com/slimsag/basisu-zig-example/blob/main/build.zig): an end-user application wants to depend on the `basisu-zig` _Zig module_ to use the Zig API:

```zig
exe.addModule("basisu-zig", b.dependency("basisu_zig").module("basisu-zig"));
```

There is _no API_ with the package manager to express that module should depend on the C library, and as a result 

```
error(link): undefined reference to symbol '_basisu_encoder_init'
```

---

Fundamentally, the problem is that the [basisu-zig](https://github.com/slimsag/basisu-zig/blob/1319c4406e1fe0ea281eba3acf67c10a7f6fa370/build.zig#L21-L27) example library has no way to express a module should link/depend on a library:

```zig
    const zig_wrapper = b.addModule("basisu-zig", .{
        .source_file = .{ .path = "src/main.zig" },
    });
    // There is *no* API on zig_wrapper (Build.Module) which would
    // allow us to link a library! This doesn't exist:
    //
    //zig_wrapper.linkLibrary(lib);
```

### Expected Behavior

There is a way for a Zig `Build.Module` to express that a static library must be linked against it, in order for it to function.

**Comment by tuket** [comment]
In addition to the lack of API to specify the need to link a C library, I think there is something else missing in order to allow making C wrappers.
If you have "main.zig" that is the root source for the module/library, and you want to expose the C API using the C header:

```zig
// main.zig
pub usingnamespace @cImport({
    @cInclude("my_c_library.h");
});
```

There is no API to add an include path for the module. For libraries we have `lib.addIncludePath()` but there isn't anything like that for modules.

**Comment by andrewrk** [comment]
I believe this is fully solved in 289ae45c1b58e952867c4fa1e246d0ef7bc2ff64
